Enhanced LaTex syntax highlighting

sesquialtera87 opened the pull request:
To the basic command and comment highlight, some new items are now recognized:

- sectioning commands are rendered differently from standard LaTex commands (`\part`, `\section` ...)
- environments name are highlighted (e.g. `\begin{itemize})`
- `\label` declarations
- `\usepackage` and `\documentclass` parameters
- math blocks are rendered as secondary language (`$...$`, `\[...\]`, `$$...$$`, `\(...\)` )

sesquialtera87 commented:
Where can I find the SpotBugs report? I see it's complaining about something, but I can't figure out what...

bobbylight commented on RSyntaxTextArea/src/main/java/org/fife/ui/rsyntaxtextarea/modes/LatexTokenMaker.flex:
```diff
@@ -65,66 +36,51 @@ import org.fife.ui.rsyntaxtextarea.*;
-
 	/**
 	 * ${inheritDoc}
 	 */
```
Can you remove this unnecessary `inheritDoc` block comment as well? Not related to your PR, but just to tidy things up a bit

sesquialtera87 commented:
I fixed some things and added tests for the new tokens.
Here a screenshot with the new highlighting.

<img width="779" height="813" alt="latex" src="https://github.com/user-attachments/assets/43f6e4d6-ff8e-4fbb-9d4d-3c00a73adb3c" />

Are rendered as independent tokens:
- options of `\documentclass` and `\usepackage` commands
- sectioning commands are differentiated
- package names, class names, environments, references
- chars in math mode is displayed as secondary language
